crates/bevy_ecs/src/schedule/graph/mod.rs

```diff
@@ -338,16 +309,15 @@ where
             }
         }
 
+        drop(stack);
+
         // remove node from subgraph
         subgraph.remove_node(root);
 
         // divide remainder into smaller SCCs
-        let mut tarjan_scc = TarjanScc::new();
-        tarjan_scc.run(&subgraph, |scc| {
-            if scc.len() > 1 {
-                sccs.push(scc.to_vec());
-            }
-        });
+        for scc in subgraph.iter_scc().filter(|scc| scc.len() > 1) {
+            sccs.push(scc);
+        }
```

Comment:
`iter_scc` already does the Vec overhead, either use `TarjanScc` directly or change this to an `Vec::extend`, which at least reduces overhead.

Reply:
Yeah that is true, I'm going to replace `iter_scc` with `for_each_scc` instead, which just wraps the `TarjanScc::run` method. I didn't want to expose the `TarjanScc` type to the rest of the crate if possible, since all that really matters is there's a way to iterate over strongly connected components; how they're computed is irrelevant to the rest of the `schedule` module.